elpis/engines/kaldi/objects/transcription.py

```diff
@@ -86,11 +106,29 @@ def _generate_inference_files(self):
     def transcribe(self, on_complete: Callable = None):
         self.status = "transcribing"
         kaldi_infer_path = self.model.path.joinpath('kaldi', 'data', 'infer')
+        gmm_decode_path = Path('/elpis/elpis/engines/kaldi/inference/gmm-decode')
+
+        # Setup logging path
+        run_log_path = self.path.joinpath('transcription.log')
         os.makedirs(f"{kaldi_infer_path}", exist_ok=True)
         dir_util.copy_tree(f'{self.path}', f"{kaldi_infer_path}")
         file_util.copy_file(f'{self.audio_file_path}', f"{self.model.path.joinpath('kaldi', 'audio.wav')}")
-        subprocess.run('sh /elpis/elpis/engines/kaldi/inference/gmm-decode-long.sh'.split(),
-                       cwd=f'{self.model.path.joinpath("kaldi")}', check=True)
+        # Copy parts of transcription process and chmod
+        dir_util.copy_tree(f'{gmm_decode_path}', f"{kaldi_infer_path.joinpath('gmm-decode')}")
+        stages = os.listdir(kaldi_infer_path.joinpath('gmm-decode'))
+        for file in stages:
+            os.chmod(kaldi_infer_path.joinpath('gmm-decode').joinpath(file), 0o774)
+        for stage in sorted(stages):
+            print(f"======== STAGE {stage} STARTING ========")
+            self.stage_status = (stage, 'in-progress', '')
+            # Setup logging
+            args = ['bash', '-c', f'touch {run_log_path}; sh {kaldi_infer_path.joinpath("gmm-decode", stage)} >> {run_log_path}']
+            p = subprocess.run(args, cwd=f'{self.model.path.joinpath("kaldi")}', check=True)
```

Comment:
I know this isn't your fault, but if you see single letter variable names please purge them with fire.